tests/stdlib/talgorithm.nim

```diff
@@ -113,3 +113,82 @@ block:
     doAssert binarySearch(moreData, 6) == -1
     doAssert binarySearch(moreData, 4711) == 4
     doAssert binarySearch(moreData, 4712) == -1
+
+# merge
+block:
+  var x = @[1, 7, 8, 11, 21, 33, 45, 99]
+  var y = @[6, 7, 9, 12, 57, 66]
+
+  let merged = merge(x, y)
+  doAssert merged.isSorted
+  doAssert merged == @[1, 6, 7, 7, 8, 9, 11, 12, 21, 33, 45, 57, 66, 99]
```

Comment:
since `sort` is already being tested, we can trust it as groundtruth; so you can use:
```nim
doAssert merged == sorted(x & y)
```
(+ elsewhere)
if you want you can have 1 explicit test though

Reply:
done